lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1189,6 +1189,65 @@ class DecomposeAtenArangeStartOp : public OpRewritePattern<AtenArangeStartOp> {
     return success();
   }
 };
+
+} // namespace
+namespace {
+// The `aten.range` op is converted to `aten.arange.start_step` op.
+class DecomposeAtenRangeOp : public OpRewritePattern<AtenRangeOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenRangeOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    // The AtenRangeOp doesn't have a step value. Therefore we set it as
+    // default value 1.
+    Value step;
+    step = rewriter.create<Torch::ConstantFloatOp>(loc,
+                                                 rewriter.getF64FloatAttr(1));
+
+    double_t end;
+
+    if (!matchPattern(op.end(), m_TorchConstantFloat(&end))) {
+        rewriter.notifyMatchFailure(op, "dim is not a valid dim");
+        return failure();
+    }
+    end++;
+    Value constEnd = rewriter.create<Torch::ConstantFloatOp>(
+        loc, rewriter.getF64FloatAttr(end));
+
+
+    rewriter.replaceOpWithNewOp<AtenArangeStartStepOp>(
+        op, op.getType(), op.start(), constEnd, step, op.dtype(), op.layout(),
+        op.device(), op.pin_memory());
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+// The `aten.range.step` op is converted to `aten.arange.start_step` op.
+class DecomposeAtenRangeStepOp : public OpRewritePattern<AtenRangeStepOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenRangeStepOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+
+    double_t end;
+
+    if (!matchPattern(op.end(), m_TorchConstantFloat(&end))) {
```

Comment:
This is looking good! I just have a suggestion for this part. I think it might be good to add lowerings for the ops

- `aten::add.int : (int, int) -> (int)`
- `aten::add.int_float : (int, float) -> (float)`

This allows you to do something like

```cpp
Value one = rewriter.create<Torch::ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
Value end;
Type endType = op.end().getType();
if (endType.isa<Torch::FloatType>())
  end = rewriter.create<Torch::AddIntFloatOp>(loc, one, op.end());
else if (endType.isa<Torch::IntType>())
  end = rewriter.create<Torch::AddIntOp>(loc, one, op.end());
else
  return rewriter.notifyMatchFailure(...);
```

This removes the dependency of having `op.end()` be `torch.constant.float` or `torch.constant.int`.

Reply:
Thanks! Actually, when I was implementing that part, I was looking for something like you mentioned above. Then figured it out there no op like that. These might be second practice for me after I have done this one.